### PR TITLE
Modify functional tests for Mint

### PR DESCRIPTION
**Pull request by poornas**
Functional tests have been modified to use streaming and standardize environment variables for running by Mint.
This PR also resolves issue of allowing Mint to use functional tests with a Nuget package SDK reference while also simultaneously allowing developers to test SDK changes locally by using a "Mint" build configuration for Functional tests  and removing it from Release build.

### CODE REVIEW

File: appveyor.yml [hunk 1]
```diff
@@ -8,9 +8,13 @@ pull_requests:
 before_build: 
  - nuget restore
  - dotnet restore
+<<<<<<< HEAD
 
 configuration: Release
 
+=======
+configuration: Release
+>>>>>>> Modify functional tests for Mint
```
Comment (NitishT) [inline appveyor.yml]: These are unintended here, I suppose.

Comment (poornas) [inline appveyor.yml]: fixed, @NitishT

**Comment by poornas** [comment]
@krisis, could you pls add your reviews

**Comment by harshavardhana** [comment]
Looks like this needs to be rebased..

**Comment by krisis** [comment]
@poornas This PR is not building successfully on Linux locally. I am unable to test it.

**Comment by poornas** [comment]
@krisis, there are 5 different build configurations -  Debug, Release, .net4.5.2, .netcore and Mint. Out of these Functional Tests project will only be built for Debug and Mint mode
 To build in Debug mode , try
$dotnet restore /p:Configuration=Debug
$dotnet build /p:Configuration=Debug

If you build the functional tests project in Mint config right now, you should get 5 errors - because they use the Nuget version which doesn't have changes related to PutObjectAsync with metadata. Otherwise it should compile.

**Comment by poornas** [comment]
@krisis, to compile the functional tests, cd into the functional tests project folder first. Rest of the commands are the same.

**Comment by krisis** [comment]
> @krisis, to compile the functional tests, cd into the functional tests project folder first. Rest of the commands are the same.
@poornas thanks that worked.

**Comment by poornas** [comment]
@NitishT , awaiting review

**Comment by NitishT** [comment]
On a side note, the dotnet installation steps in the readme are not enough for Ubuntu 17.04

Following needs to be added to readme 
- https://www.microsoft.com/net/core#linuxdebian
- https://github.com/dotnet/cli/issues/2018#issuecomment-252481343

**Comment by harshavardhana** [comment]
> On a side note, the dotnet installation steps in the readme are not enough for Ubuntu 17.04
> 
> Following needs to be added to readme
> 
> https://www.microsoft.com/net/core#linuxdebian
> dotnet/cli#2018 (comment)

Current steps are only for `xenial` i.e 16.04 LTS  - 17.04 is not an TLS release so it was decided to not add any additional docs for them.

**Comment by NitishT** [comment]
> Current steps are only for xenial i.e 16.04 LTS - 17.04 is not an LTS release so it was decided to not add any additional docs for them.

I see. Then we can change the heading here accordingly - https://github.com/minio/minio-dotnet/blob/master/README.md#on-linux-ubuntu-1604-and-above

**Comment by poornas** [comment]
@NitishT , updated the documentation to specify Linux instructions only for Xenial 16.04

**Comment by NitishT** [comment]
@poornas any ideas on why `continuous-integration/appveyor/branch` is not running

**Comment by poornas** [comment]
@NitishT , this seems to be a repo setting issue - http://help.appveyor.com/discussions/problems/3612-github-showing-multiple-checks. @deekoder, can you check?

**Comment by harshavardhana** [comment]
Just ammend the commit and push again. It will build again some times build might have missed the API.

**Comment by poornas** [comment]
@harshavardhana, have amended commit and pushed a few times already. Both PRs on this repo fail to report back Appveyor status.Pushed once more with amended commit.

**Comment by harshavardhana** [comment]
> @harshavardhana, have amended commit and pushed a few times already. Both PRs on this repo fail to report back Appveyor status.Pushed once more with amended commit.

Actually it was my mistake i had branch build enabled on a PR it is removed now and should be fine to proceed and merge.

**Comment by poornas** [comment]
@harshavardhana , sounds good - thanks!

**Pull request merged**